### PR TITLE
fix(pure-utils): Add option not to parse path params into numbers

### PR DESCRIPTION
**Pull request by TomiHiltunen**
This allows developer to have the router not do automatic string to number parsing on some routes.
Numeric strings with too big values will not parse correctly to integers and this fix allows
bypassing the default behavior of parsing strings into integers.

BREAKING CHANGE: None

https://github.com/faceyspacey/redux-first-router/issues/242

### CODE REVIEW

**Comment by hedgepigdaniel** [comment]
Is it necessary to make the number parsing work for really large numbers?

As I see it, 99% of the problem is that developers who do not want or expect their url strings to be parsed as numbers currently tend to get burned: once when they get a number instead of a string (possibly in a rare case in production where a user generated path segment happens to be a number), and then again when the number parsing doesn't work for really large numbers.

I think the best solution is for this library not to parse numbers in the first place. Having an option as its been done here is a simple way of providing this option without introducing a breaking change.

Basically, having number parsing built in is a neat bonus, but having strings by default sometimes come out as numbers depending on their value and even sometimes an incorrectly expressed number causes serious bugs in applications.

**Comment by hedgepigdaniel** [comment]
This PR should probably be made into the `rudy` branch, seeing as that's the latest release before the big refactor

**Comment by adamdicarlo** [comment]
We just ran into a parameter, which is a randomly generated id, not working because it was `'400960e3'`... notice how it ends with `e3` which looks like exponential notation. It gets stored as the number `400960000`; even if you convert that value to a string, it doesn't match the string `'400960e3'`.

**Comment by ScriptedAlchemy** [comment]
Closing as its already fixed in rudy
